src/screens/Profile/Profile.tsx

```diff
@@ -1,35 +1,63 @@
-import React from 'react'
-import { View, Image, TouchableOpacity, Linking, FlatList } from 'react-native'
+import React, { useEffect } from 'react'
+import {
+  View,
+  Image,
+  TouchableOpacity,
+  Linking,
+  FlatList,
+  ActivityIndicator
+} from 'react-native'
+import { useDispatch } from 'react-redux'
+import PodcastItem from './components'
+import { useAppSelector } from 'src/hooks'
+import { RootState } from 'src/common/types'
+import { resetUser, loadPodcasts } from 'src/store/actions'
 import { Heading, HeadingType, PlainText } from 'src/components'
-import { Podcast, User } from 'src/common/types'
-import { MockedUser, MockedPodcasts } from './mockedData'
+import { Podcast } from 'src/common/types'
+import { DataStatus } from 'src/common/enums'
 import Check from 'src/assets/images/checkMark.svg'
 import AtMark from 'src/assets/images/atMark.svg'
 import LogOut from 'src/assets/images/iconmonstr-log-out-16.svg'
-import PodcastItem from './components'
 import styles from './styles'
 
-type Props = {
-  podcasts: Podcast[]
-  user: User
-}
+const Profile: React.FC = () => {
+  const dispatch = useDispatch()
 
-const userPageInfo: Props = {
-  podcasts: MockedPodcasts,
-  user: MockedUser
-}
+  const handleLogOut = (): void => {
+    dispatch(resetUser())
+  }
 
-const UserPage: React.FC<Props> = () => {
-  const handleLogOut = () => {
-    return 0
+  const { user, podcasts, dataStatus } = useAppSelector(
+    ({ auth, user }: RootState) => ({
+      user: auth.user,
+      podcasts: user.podcasts,
+      dataStatus: user.dataStatus
+    })
+  )
+
+  useEffect(() => {
+    dispatch(loadPodcasts(Number(user?.id)))
+  }, [user?.id])
+
+  if (!user) {
+    return null
   }
 
-  const handleOpenMail = async () => {
-    await Linking.openURL(
-      `mailto://${userPageInfo.user.email}}&subject=abcdefg&body=body`
+  const hasUser = Boolean(user)
```

Comment:
I'm not sure if need this. Because without user data we would not have got access beyound the 'Login' screen. There is a guarantee that user data exists always on the 'Profile' screen.